lib/rubocop/cop/lint/self_assignment.rb

```diff
@@ -26,6 +26,15 @@ class SelfAssignment < Base
           gvasgn: :gvar
         }.freeze
 
+        def on_send(node)
+          case node.method_name
+          when :[]=
+            handle_key_assignment(node) if node.arguments.size == 2
+          when /=\z/
+            handle_attribute_assignment(node) if node.arguments.size == 1
+          end
```

Comment:
Can you tweak to the below?

```suggestion
          if node.method?(:[]=)
            handle_key_assignment(node) if node.arguments.size == 2
          elsif node.assignment_method?
            handle_attribute_assignment(node) if node.arguments.size == 1
          end
```